### DISCUSSION TITLE
Starting the Emulator from a Bash script doesn't work

### DISCUSSION THREAD

**Discussion by SirNibble** [prirun/p50em#22] [General]
Pi4 Linux 5.4.83-v7l+ #1379 SMP Mon Dec 14 13:11:54 GMT 2020 armv7l

```
#! /bin/bash

./em -boot 14114 -tport 9920 -nport 9120 -cpuid 9955ii
```

Gives ..
```
pi@sandsend:~/PRIME/GWHB $ ./boot.sh
[Prime Emulator ver ebf5511 May  5 2020]
[Copyright (C) 2005-2019 Jim Wilcoxson prirun@gmail.com]

Fatal error
instruction #0 at 0/0  ^000000^
A='0/0  B='0/0  L='0/0  X='0/0 K=0 [16SF000/0 ]
owner=0 , modals=0 [I0N]
Check error.log for more information

pi@sandsend:~/PRIME/GWHB $ cat error.log
 error opening /dev/tty: No such device or address
pi@sandsend:~/PRIME/GWHB $
```

Am I missing something? (And would this be more appropiate as a GitHub Issue?)

**Comment by hashbackup** [comment]
In emdev.h, devasr() is coded for a real keyboard and screen aka
/dev/tty.  /dev/tty can be both read and written using one file
descriptor, whereas stdin and stdout are separate descriptors.  To run
the emulator from a shell, another file descriptor would be needed and
devasr would have to be changed.  It's not hard work if you want to
have a go at it.  Or you can use screen like the public emulators.

**Comment by SirNibble** [comment]
Perfect, thanks. 

Updated my script to

```
#! /bin/bash

screen ./em -boot 14114 -tport 9920 -nport 9120 -cpuid 9955ii
```

and no more forgetting to set parameters!

I'd seen the examples but, tbh, hadn't really paid attention to the 'screen' bit as I'm interacting via multiple PuTTY sessions. I'd forgotten how useful that functionality is on real terminals.

My 'C' isn't brilliant, but I may have a fettle one day,
